test/common/stream_info/test_util.h

```diff
@@ -38,6 +38,7 @@ class TestStreamInfo : public StreamInfo::StreamInfo {
   const absl::optional<std::string>& responseCodeDetails() const override {
     return response_code_details_;
   }
+  void setResponseCode(uint32_t code) override { response_code_ = code; }
```

Comment:
I would be in favor of moving this small addition to another pull request if others think that's better.